client/lib/media/utils/get-file-uploader.js

```diff
@@ -3,15 +3,53 @@ import wpcom from 'calypso/lib/wp';
 
 const debug = debugFactory( 'calypso:media' );
 
+const getMime = ( f ) => f?.type || f?.file?.type || f?.fileContents?.type || '';
+
+const toTusFile = ( input ) => {
+	if ( typeof File !== 'undefined' && input instanceof File ) {
+		return input;
+	}
+	if ( typeof Blob !== 'undefined' && input instanceof Blob ) {
+		// Bare blob, add a name 'upload'
+		return new File( [ input ], 'upload', { type: input.type || 'application/octet-stream' } );
+	}
+	if ( input?.fileContents instanceof Blob ) {
+		const blob = input.fileContents;
+		const name = input.fileName || input.name || 'upload';
+		return new File( [ blob ], name, {
+			type: blob.type || input.type || 'application/octet-stream',
+		} );
+	}
+	// Unknown, return what they gave us
+	return input;
+};
```

Comment:
This code is brand new, isn't it? It's not in the original code. There we passed the `file` object directly to the `tus-js-client` `Upload` constructor. At first sight it's surprising that it ever worked at all 🙂

Reply:
Yes, exactly!

Reply:
So the TUS uploader never really worked because it was getting a `file` object it didn't understand? Interesting 🙂